state/azure/cosmosdb/cosmosdb.go

```diff
@@ -179,9 +221,21 @@ func (c *StateStore) Delete(req *state.DeleteRequest) error {
 		return err
 	}
 
-	selfLink := fmt.Sprintf("dbs/%s/colls/%s/docs/%s", c.db.Id, c.collection.Id, req.Key)
+	partitionKey := populatePartitionMetadata(req.Key, req.Metadata)
+	options := []documentdb.CallOption{documentdb.PartitionKey(partitionKey)}
 
-	options := []documentdb.CallOption{documentdb.PartitionKey(req.Key)}
+	items := []CosmosItem{}
+	_, err = c.client.QueryDocuments(
+		c.collection.Self,
+		documentdb.NewQuery("SELECT * FROM ROOT r WHERE r.id=@id", documentdb.P{Name: "@id", Value: req.Key}),
```

Comment:
The delete api takes in a self link to the item to delete (rather than the key) and I don't think it's possible derive get this without querying it.  All the examples show this method.  The original code which formatted the self link based on a format string did not work.  The actual self link has the format string values hashed and if you provide them unhashed it doesn't do it for you.  For example this 
dbs/a/colls/b/docs/c 

might turn into something like 

dbs/dL3mAA==/colls/dE5mAJ-kY9w=/docs/dL9mAJ-kY7xPBQAAAAABBA==/

>> If we have to query use count(*) so that all the data is not returned only 1 integer is returned and then if the value is 0, it doesnt exist.
I'm not sure I understand what you're saying here, our Delete interface expects a single item for a single key.

Reply:
thanks for explaining, that sounds fine.